src/inmanta/module.py

```diff
@@ -427,9 +452,60 @@ def get_module(
         module_name: str = self._get_module_name(module_spec)
         installed: Optional[TModule] = self.get_installed_module(project, module_name)
         if installed is None and install:
-            return self.install(project, module_spec)
+            installed = self.install(project, module_spec)
+
         return installed
 
+    @abstractmethod
+    def log_pre_install_information(self, project: "Project", module_name: str) -> Dict[str, "Version"]:
+        """
+        Display information about this module's installation before the actual installation.
+
+        :param project: The project associated with the module.
+        :param module_name: The module's name.
+        """
+        raise NotImplementedError("Abstract method")
+
+    @abstractmethod
+    def log_post_install_information(
+        self, project: "Project", module_name: str, modules_pre_install: Dict[str, "Version"]
+    ) -> None:
+        """
+        Display information about this module's installation after the actual installation.
+
+        :param project: The project associated with the module.
+        :param module_name: The module's name.
+        """
+        raise NotImplementedError("Abstract method")
+
+    @staticmethod
+    def log_snapshot_difference(modules_pre_install: Dict[str, "Version"]) -> None:
+        """
+        Logs all inmanta modules currently installed (in alphabetical order) and their version.
+        For each module, the prefix gives some context:
+            - "+" means this module is newly installed
+            - "~" means this module was previously installed but a new version has been installed
+            - " " means this module was already installed and left untouched
+        """
+        version_snapshot = env.PythonWorkingSet.get_packages_in_working_set(inmanta_modules_only=True)
+        set_pre_install = set(modules_pre_install.items())
+        set_post_install = set(version_snapshot.items())
+        updates_and_additions = set_post_install - set_pre_install
+
+        if version_snapshot:
+            out = ["Snapshot of modules versions post-install:"]
+            for package_name, package_version in sorted(version_snapshot.items()):
+                prefix = " "
+                suffix = ""
+                if package_name not in modules_pre_install.keys():
+                    prefix = "+"
+                elif package_name in [elmt[0] for elmt in updates_and_additions]:
+                    prefix = "~"
+                    suffix = f" (was previously {modules_pre_install[package_name]})"
+
+                out.append(prefix + package_name + ": " + str(package_version) + suffix)
+            LOGGER.info("\n".join(out))
```

Comment:
>  This implementation logs a message each time a module is installed. At the end of the installation we also need to provide an overview of all the modules that are installed, because some modules may have been installed as a transitive dependency of a certain module. The overview should be logged in the log level INFO. The messages for the installation of a specific package in DEBUG. That way the user can decide whether he/she wants to see all the details or not.

I logged this overview at the info level because of this :point_up:

Reply:
Oh nevermind, I missed your other comment about addind a proper overview on `inmanta export` or `inmanta compile`